fv/wireguard_test.go

```diff
@@ -0,0 +1,455 @@
+				out, _ := felix.ExecOutput("wg")
+				matches := xferRegExp.FindStringSubmatch(out)
+				Expect(matches).NotTo(BeNil())
+				rcvd[i] = matches[1]
+				sent[i] = matches[2]
+			}
+			Expect(rcvd[0]).To(Equal(sent[1]))
+			Expect(rcvd[1]).To(Equal(sent[0]))
+		})
+	})
+
+	Context("with Wireguard disabled", func() {
+		BeforeEach(func() {
+			felixPID := felixes[0].GetFelixPID()
+
+			enableDisableWireguard(client, false)
+
+			// Wait for felix to restart.
+			Eventually(felixes[0].GetFelixPID, "5s", "100ms").ShouldNot(Equal(felixPID))
+		})
+
+		It("the Wireguard device shouldn't exist", func() {
+			for _, felix := range felixes {
+				Eventually(func() string {
+					out, _ := felix.ExecOutput("ip", "link", "show", wireguardInterfaceNameDefault)
+					return out
+				}, "10s", "100ms").Should(BeEmpty())
+			}
+		})
+
+		It("the Wireguard rule-entry/route-table shouldn't exist", func() {
+			for i, felix := range felixes {
+				exists := wireguardRoutingRuleRouteEntryExists(felix, routingRuleRegExp, routeEntries[i])
+				Eventually(exists).Should(BeFalse(), "wireguard rule unexpectedly present")
+			}
+		})
+
+		It("v3 node resource doesn't contain public-keys", func() {
+			for _, felix := range felixes {
+				Eventually(func() string {
+					node, err := client.Nodes().Get(context.Background(), felix.Hostname, options.GetOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					return node.Status.WireguardPublicKey
+				}, "5s", "100ms").Should(BeEmpty())
+			}
+		})
+
+		It("workload connectivity remains but uses un-encrypted tunnel", func() {
+			cc.ExpectSome(w[0], w[1])
+			cc.ExpectSome(w[1], w[0])
+			cc.CheckConnectivity()
+
+			for _, felix := range felixes {
+				Eventually(func() string {
+					// No tunnel implies un-encrypted communication.
+					out, _ := felix.ExecOutput("wg")
+					return out
+				}, "10s", "100ms").Should(BeEmpty())
+			}
+		})
+	})
+})
+
+var _ = infrastructure.DatastoreDescribe("WireGuard-Unsupported", []apiconfig.DatastoreType{apiconfig.EtcdV3, apiconfig.Kubernetes}, func(getInfra infrastructure.InfraFactory) {
+	var (
+		infra   infrastructure.DatastoreInfra
+		felixes []*infrastructure.Felix
+	)
+
+	BeforeEach(func() {
+		// Setup a single node cluster.
+		const nodeCount = 1
+
+		infra = getInfra()
+		felixes, _ = infrastructure.StartNNodeTopology(nodeCount, wireguardTopologyOptions(), infra)
+
+		// Install a default profile that allows all ingress and egress, in the absence of any Policy.
+		infra.AddDefaultAllow()
+
+		felixes[0].TriggerDelayedStart()
+	})
+
+	AfterEach(func() {
+		if CurrentGinkgoTestDescription().Failed {
+			felixes[0].Exec("ip", "link")
+			felixes[0].Exec("wg")
+		}
+
+		felixes[0].Stop()
+
+		if CurrentGinkgoTestDescription().Failed {
+			infra.DumpErrorData()
+		}
+		infra.Stop()
+	})
+
+	It("no Wireguard device exists", func() {
+		Eventually(func() string {
+			out, _ := felixes[0].ExecOutput("ip", "link", "show", wireguardInterfaceNameDefault)
+			return out
```

Comment:
You are comparing a Zero value, so need to check for error.  In general it's safer to just always check the error value as well.

Reply:
yeah, I added checking error for all `ExecOutput`, testing now.